### PR TITLE
Polish Gradle warning summary

### PR DESCRIPTION
**Pull request by oehme**
These warnings are not just about deprecations, but will
include other kinds of warnings in the future. Calling them
Gradle warnings also makes it clear that we are not talking
about Java compiler warnings for instance.

Mentioning that this may break is redundant with what the
individual warnings are already saying and might actually
be wrong. E.g. if there is a single deprecation for Gradle
6.0, then saying "this may break in 5.0" is incorrect. It
also sounds scarier than it needs to be.

### CODE REVIEW

**Comment by oehme** [comment]
@eriwen I just saw you also worked on this. You might just want to merge this wording change into your PR.

**Comment by bmuschko** [comment]
@eriwen @oehme Which PR are we going to merge? Can you please close one of them? Happy to do the review for the remaining.

**Comment by oehme** [comment]
@eriwen wanted to combine them, taking the proposed warning here into his PR.

**Comment by eriwen** [comment]
@oehme I think the message "there are deprecation warnings that may make this build incompatible with Gradle 5.0" is important enough that I hesitate to remove a mention of deprecation warnings altogether.

**Comment by oehme** [comment]
Your call, though I find the message currently proposed in the other PR too wordy to display after every build:

"Deprecated Gradle API and/or features were used in this build, making it incompatible with Gradle 5.0. Read documentation on configuring warning logging at ..."

If you want to stick with the "incompatible" part, then maybe shorten it down to

"Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0. See ... ."

**Comment by eriwen** [comment]
Sounds good @oehme. I'd like to take your last suggestion here. This PR is superseded by #4066
